### PR TITLE
ipc4: using xtensa-build-all.sh to build ipc4 firmware

### PR DESCRIPTION
**Pull request by bardliao**
This PR allow us to use xtensa-build-all.sh to build ipc4 firmware.

### CODE REVIEW

File: src/arch/xtensa/configs/tgphipc4_defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CONFIG_TIGERLAKE=y
+CONFIG_RIMAGE_SIGNING_SCHEMA="tgl-cavs"
```
Comment (marc-hb) [inline src/arch/xtensa/configs/tgphipc4_defconfig:2]: Does IPC4 really requires a new signing schema?

**Review by marc-hb — COMMENTED** [review]
Copy/paste/diverge 95% of an existing configuration does not seem scalable. How many platforms will support both IPC3 and IPC4?

Have you considered a one-line `src/arch/xtensa/configs/override/IPC4.config instead?

File: src/arch/xtensa/configs/tgphipc4_defconfig (hunk 1 above)
Comment (lgirdwood) [reply src/arch/xtensa/configs/tgphipc4_defconfig:2]: yes.

**Comment by lgirdwood** [comment]
@bardliao can you fixup @marc-hb comment is a subsequent PR. Thanks
@marc-hb fwiw, this branch is all WIP code - aim is to allow a **working** base for IPC4 work whilst developers upstream their own features.

**Comment by lgirdwood** [comment]
@bardliao thanks, I did rebase last night. I the whole patch missing or just parts, no sure why rebase  would miss stuff ?

**Comment by bardliao** [comment]
> @bardliao thanks, I did rebase last night. I the whole patch missing or just parts, no sure why rebase would miss stuff ?

@lgirdwood I created a new PR https://github.com/thesofproject/sof/pull/4338 to address @marc-hb 's comments.

**Pull request merged**